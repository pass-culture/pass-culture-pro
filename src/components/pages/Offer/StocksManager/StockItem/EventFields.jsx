import moment from 'moment'
import PropTypes from 'prop-types'
import React, { Component, Fragment } from 'react'

import { DateField, HiddenField, TimeField } from 'components/layout/form'

export class EventFields extends Component {
  /*
  componentDidUpdate(prevProps) {
    this.handleCrossingEndDatetime()
  }

  handleCrossingEndDatetime = () => {
    const {
      dispatch,
      formBeginningDatetime,
      formEndDatetime,
      stockFormKey,
    } = this.props

    if (formEndDatetime >= formBeginningDatetime) {
      return
    }

    const endDatetime = getDatetimeOneDayAfter(formBeginningDatetime)

    dispatch(mergeForm(`stock${stockFormKey}`, { endDatetime }))
  }
  */

  render() {
    const { beginningMinDate, readOnly, stocks, tz } = this.props
    const highlightDates = (stocks || []).map(stock => stock.beginningDatetime)

    return (
      <Fragment>
        <td>
          <DateField
            highlightDates={highlightDates}
            minDate={beginningMinDate || moment()}
            name="beginningDatetime"
            readOnly={readOnly}
            required
            title="Date"
          />
        </td>
        <td>
          <TimeField
            name="beginningTime"
            patchKey="beginningDatetime"
            readOnly={readOnly}
            required
            title="Heure"
            tz={tz}
          />
        </td>
        <td>
          <HiddenField name="endDatetime" />
          <TimeField
            name="endTime"
            patchKey="endDatetime"
            readOnly={readOnly}
            required
            title="Heure de fin"
            tz={tz}
          />
        </td>
      </Fragment>
    )
  }
}

EventFields.defaultProps = {
  beginningMinDate: null,
  stocks: null,
}

EventFields.propTypes = {
  beginningMinDate: PropTypes.string,
  dispatch: PropTypes.func.isRequired,
  stocks: PropTypes.array,
}

export default EventFields
