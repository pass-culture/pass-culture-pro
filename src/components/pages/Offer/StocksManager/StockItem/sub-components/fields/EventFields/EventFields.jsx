import moment from 'moment'
import PropTypes from 'prop-types'
import React, { Fragment } from 'react'

import DateField from 'components/layout/form/fields/DateField/DateField'
import TimeField from 'components/layout/form/fields/TimeField'

const EventFields = ({ beginningMinDate, readOnly, stocks, timezone }) => {
  const highlightDates = (stocks || []).map(stock => stock.beginningDatetime)

  return (
    <Fragment>
      <td>
        <DateField
          highlightDates={highlightDates}
          minDate={beginningMinDate || moment()}
          name="beginningDatetime"
          readOnly={readOnly}
          required
          timezone={timezone}
          title="Date"
        />
      </td>
      <td>
        <TimeField
          name="beginningTime"
          patchKey="beginningDatetime"
          readOnly={readOnly}
          required
          title="Heure"
        />
      </td>
    </Fragment>
  )
}

EventFields.defaultProps = {
  beginningMinDate: null,
  readOnly: true,
  stocks: null,
  timezone: null,
}

EventFields.propTypes = {
  beginningMinDate: PropTypes.string,
  readOnly: PropTypes.bool,
  stocks: PropTypes.arrayOf(PropTypes.shape()),
  timezone: PropTypes.string,
}

export default EventFields
