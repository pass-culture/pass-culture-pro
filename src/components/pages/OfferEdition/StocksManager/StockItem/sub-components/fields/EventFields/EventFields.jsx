import moment from 'moment'
import PropTypes from 'prop-types'
import React, { Fragment } from 'react'

import DateField from '../../../../../../../layout/form/fields/DateField'
import HiddenField from '../../../../../../../layout/form/fields/HiddenField'
import TimeField from '../../../../../../../layout/form/fields/TimeField'

const EventFields = ({ beginningMinDate, readOnly, stocks, timezone }) => {
  const highlightDates = (stocks || []).map(stock => stock.beginningDatetime)

  return (
    <Fragment>
      <td>
        <DateField
          highlightDates={highlightDates}
          minDate={beginningMinDate || moment()}
          name="beginningDatetime"
          readOnly={readOnly}
          required
          timezone={timezone}
          title="Date"
        />
      </td>
      <td>
        <TimeField
          name="beginningTime"
          patchKey="beginningDatetime"
          readOnly={readOnly}
          required
          title="Heure"
        />
      </td>
      <td>
        <HiddenField name="endDatetime" />
        <TimeField
          name="endTime"
          patchKey="endDatetime"
          readOnly={readOnly}
          required
          title="Heure de fin"
        />
      </td>
    </Fragment>
  )
}

EventFields.defaultProps = {
  beginningMinDate: null,
  readOnly: true,
  stocks: null,
  timezone: null,
}

EventFields.propTypes = {
  beginningMinDate: PropTypes.string,
  readOnly: PropTypes.bool,
  stocks: PropTypes.arrayOf(PropTypes.shape()),
  timezone: PropTypes.string,
}

export default EventFields
