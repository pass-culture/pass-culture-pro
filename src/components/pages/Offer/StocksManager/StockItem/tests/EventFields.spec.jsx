import React from 'react'
import { shallow } from 'enzyme'

import EventFields from '../EventFields'

describe('src | components | pages | Offer | StockItem | EventFields', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const initialProps = {
        beginningMinDate: '2019-03-29T01:56:55.610Z',
        dispatch: jest.fn(),
        parseFormChild: jest.fn(),
      }

      // when
      const wrapper = shallow(<EventFields {...initialProps} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
