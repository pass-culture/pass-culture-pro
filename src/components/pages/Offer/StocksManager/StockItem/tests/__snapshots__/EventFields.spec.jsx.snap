// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Offer | StockItem | EventFields snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventFields
    dispatch={[Function]}
    formBeginningDatetime={null}
    formBookingLimitDatetime={null}
    formEndDatetime={null}
    parseFormChild={[Function]}
    stocks={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
